crates/revm/src/context.rs

```diff
@@ -110,38 +106,10 @@ impl<EXT, DB: Database> Host for Context<EXT, DB> {
     }
 
     fn block_hash(&mut self, number: U256) -> Option<B256> {
-        let block_number = as_usize_saturated!(self.env().block.number);
-        let requested_number = as_usize_saturated!(number);
-
-        let Some(diff) = block_number.checked_sub(requested_number) else {
-            return Some(B256::ZERO);
-        };
-
-        // blockhash should push zero if number is same as current block number.
-        if diff == 0 {
-            return Some(B256::ZERO);
-        }
-
-        if diff <= BLOCK_HASH_HISTORY {
-            return self
-                .evm
-                .block_hash(number)
-                .map_err(|e| self.evm.error = Err(e))
-                .ok();
-        }
-
-        if self.evm.journaled_state.spec.is_enabled_in(PRAGUE) && diff <= BLOCKHASH_SERVE_WINDOW {
```

Comment:
only this PRAGUE branch should be removed, the rest is still valid (diff == 0, diff <= HISTORY...)

Reply:
are you sure? from the EIP:
> The `BLOCKHASH` opcode semantics remains the same as before.


https://github.com/bluealloy/revm/blob/eed27d92266776d653a1d08c46ffa9699d64d527/crates/revm/src/context.rs#L108-L113

Reply:
Yes, this is the before:
https://github.com/bluealloy/revm/blob/688c36ca4525cc44aa7c547b7b0f22a9490c4f2f/crates/interpreter/src/instructions/host.rs#L106-L123

Reply:
oh, my bad, fixing